### PR TITLE
Add 7.Lifecycle

### PR DESCRIPTION
**Pull request by neo**

### CODE REVIEW

File: 7.Lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+**Jam3 Lessons - React**
+
+# Lifecycle
+
+> Each component has several "lifecycle methods" that you can override to run code at particular times in the process.
+Methods prefixed with will are called right before something happens, and methods prefixed with did are called right after something happens.
```
Comment (jeremenichelli) [inline 7.Lifecycle.md:6]: I think it sounds better with this changes:

`Each component has several _methods_ that you can override to run code at particular moments of a component's _life_.`

For the second paragraph I would delete it and list all the lifecycle callbacks briefly explaining when they are executed so it actually serves a quick reference for any developer.

Let me know what you think @neo.

File: 7.Lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+We could ask a React component to do certain things at certain stage of the lifecycle:
+
+```JSX
```
Comment (jeremenichelli) [inline 7.Lifecycle.md]: lowercase for language hints

File: 7.Lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class Welcome extends React.Component {
+  // ...
```
Comment (jeremenichelli) [inline 7.Lifecycle.md]: Remove unnecessary comment

File: 7.Lifecycle.md (hunk 1 above)
Comment (neo) [reply 7.Lifecycle.md:6]: Quotes are all copied from the React docs, if you think we should describe on our own, I could rewrite it. I just thought that they wrote it really accurate and very easy to understand, I can't find anything to change.

Comment (jeremenichelli) [reply 7.Lifecycle.md:6]: yeah, cool, let's leave it like this for now. Thank you :)

Comment (neo) [reply 7.Lifecycle.md:6]: Just they wrote it so good... 😳

**Pull request merged**